mozilla/network-pulse-api
# POST autocompleted creators for entries

This PR does two things:
1. Adds functionality to auto-create creator objects when profiles are created and creates creator objects for existing profiles.
2. Changes POST /entries to accept existing creators and create new ones if they don't exist.

I also now force JSON content-type for entries and as a result had to fix some tests for that.

Another small change I made was how we access `request` from the context...we previously accessed it as if it were an object but the context is actually a dict...that's what caused https://github.com/mozilla/network-pulse-api/issues/233.

### To test that creators have profiles:
1. Check the db for a profile you like
2. Run `python manage.py migrate`
3. Check the db again and make sure the `creators_creator` table has an entry with `profile_id` set to your selected profile's id

### To test posting related_creators:
1. Go to `test.example.com:8000/api/pulse/entries/` and go to the bottom of the page
2. Open your console and type in `var x = new XMLHttpRequest(); x.open('GET', 'http://test.example.com:8000/api/pulse/nonce'); x.onreadystatechange = function() { if(x.readyState === 4) { console.log(x.responseText); } }; x.send();` and hit Enter
3. Copy and paste the `nonce` and `csrf_token` into the empty JSON payload auto-generated for you at the bottom of the page. Change `csrf_token` to `csrfmiddlewaretoken`.
4. Fill out at least title. Set `related_creators` property to an array of objects with either an `id` key or a `name` key. The `id` key should be an existing creator `id` (check from the admin). Hit Send.
5. Make sure it succeeds and verify the actual entry.
